tests/lib/hardware/openstack_sdk.py

```diff
@@ -226,8 +226,23 @@ def boot_nodes(self, masters: int, workers: int, offset: int = 0):
         for t in threads:
             t.join()
 
-    def destroy(self):
-        super().destroy()
+    def destroy(self, skip=False):
+        super().destroy(skip=skip)
+
+        if skip:
+            if self._router_private:
+                logger.warning(f"Leaving router {self._router_private.name}")
+            if self._subnet_private:
+                logger.warning(f"Leaving subnet {self._subnet_private.name}")
+            if self._network_private:
+                logger.warning(f"Leaving network {self._network_private.name}")
+            if self._security_group:
+                logger.warning(
+                    f"Leaving security group {self._security_group.name}")
+            if self._keypair:
+                logger.warning(f"Leaving keypair {self._keypair.name}")
```

Comment:
What about the nodes and disks?

Reply:
They get mentioned from the super method on hardware_base.